apps/pkeyutl.c

```diff
@@ -493,3 +544,35 @@ static int do_keyop(EVP_PKEY_CTX *ctx, int pkey_op,
     }
     return rv;
 }
+
+static void print_supported_kdfs(void) {
+    const struct kdf_info *kdf = supported_kdfs;
+    BIO_printf(bio_err, "Supported KDFs:\n");
+    while (kdf->name) {
+        BIO_printf(bio_err, "  %-10s %s\n", kdf->name, kdf->description);
+        kdf++;
+    }
+}
+
+static void print_kdf_options(const char *kdfalg) {
+    const struct kdf_info *kdf = supported_kdfs;
+    const struct pkey_option *option;
+    while (kdf->name) {
+        if (!strcmp(kdfalg, kdf->name)) {
+            break;
+        }
+        kdf++;
+    }
+
+    if (!kdf->name) {
+        /* KDF not listed in info entries */
+        return;
+    }
+
+    BIO_printf(bio_err, "Supported options of the \"%s\" KDF:\n", kdfalg);
+    option = kdf->pkey_options;
+    while (option->keys) {
+        BIO_printf(bio_err, "  %-20s %s%s\n", option->keys, option->description, option->mandatory ? "; mandatory argument" : "");
```

Comment:
I would suggest to execute `env INDENT_PROFILE=util/indent.pro indent apps/pkeyutl.c`. It won't get everything right, e.g. it fails to get `STACK_OF(OPENSSL_STRING) *pkeyopts = NULL;` right, but it will show how vertical alignment for subroutine arguments is supposed to look...